src/pkgs/files/login-inner.nix

```diff
@@ -16,12 +16,32 @@ writeScript "login-inner" ''
   shift
 
   [ "$#" -gt 0 ] || echo "Sourcing Nix environment..."
-  . ${instDir}/${packageInfo.nix}/etc/profile.d/nix.sh
+  . ${packageInfo.nix}/etc/profile.d/nix.sh
 
-  if [ ! -e $HOME/.nix-profile/etc/ssl/certs/ca-bundle.crt ]; then
-    if [ -e ${packageInfo.cacert} ]; then
-      export NIX_SSL_CERT_FILE=${packageInfo.cacert}
-    fi
+  if [ -e ${instDir}/etc/UNINTIALISED ]; then
+    export NIX_SSL_CERT_FILE=${packageInfo.cacert}
+
+    echo "Installing and updating nix-channels..."
+    ${packageInfo.nix}/bin/nix-channel --add https://nixos.org/channels/nixos-19.03 nixpkgs
+    ${packageInfo.nix}/bin/nix-channel --add https://github.com/t184256/nix-on-droid-bootstrap/archive/master.tar.gz nix-on-droid
+    ${packageInfo.nix}/bin/nix-channel --update
+
+    echo "Installing nix-on-droid.basic-environment..."
+    ${packageInfo.nix}/bin/nix-env -iA nix-on-droid.basic-environment
```

Comment:
I'm against imperative PM using nix-env. Please accomplish the same in reverse: drop in the default config here and build the first home-manager generation, which, in turn, would install basic-environment.

Reply:
But in that way we would force the installation of home-manager what we not intended to do.